desc/magnetic_fields.py

```diff
@@ -225,8 +422,8 @@ class ToroidalMagneticField(MagneticField):
     _io_attrs_ = MagneticField._io_attrs_ + ["_B0", "_R0"]
 
     def __init__(self, B0, R0):
-        assert np.isscalar(B0), "B0 must be a scalar"
-        assert np.isscalar(R0), "R0 must be a scalar"
+        assert float(B0) == B0, "B0 must be a scalar"
```

Comment:
if we're making this change we should also explicitly cast the values to floats before assigning so that they're actually scalars not size 1 arrays